Grails 3.3.5: resources.groovy not evaluated in Integration Tests

When trying to autowire a bean in an integration test that was defined in ``resources.groovy``, the app fails to start in the test environment. When defining the bean in a plugin's ``doWithSpring`` closure instead, it works as expected. When trying to deliberately throw a ``RuntimeException`` in ``resources.groovy`` it turns out that the application does not run into that exception before failing to inject the bean, which I assume means that ``resources.groovy`` does not even get evaluated in that environment.

### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a new Grails plugin with Grails 3.3.5: ``grails create-plugin bean.test.bean-test``
2. Add an arbitrary class to [``src/main/groovy``](https://github.com/miffels/catfood/blob/master/bean-test/src/main/groovy/bean/test/beans/DummyDependency.groovy), e.g. ``DummyDependency``
3. Add a bean definition in [``grails-app/conf/spring/resources.groovy``](https://github.com/miffels/catfood/blob/master/bean-test/grails-app/conf/spring/resources.groovy)
4. Add an [integration test that is trying to autowire the bean](https://github.com/miffels/catfood/blob/master/bean-test/src/integration-test/groovy/bean/test/DummySpecification.groovy)
5. Run the integration test: ``./gradlew integrationTest``

### Expected Behaviour

The bean should be injected and the test should succeed

### Actual Behaviour

Before the test suite is executed, the application crashes with the following ``UnsatisfiedDependencyException``:

```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'bean.test.DummySpecification': Unsatisfied dependency expressed through field 'dummyDependency'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'bean.test.beans.DummyDependency' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.spockframework.runtime.model.FieldMetadata(line=13, name=dummyDependency, ordinal=0)}
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:588)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1268)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireBeanProperties(AbstractAutowireCapableBeanFactory.java:386)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:118)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:83)
	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:230)
	at org.spockframework.spring.SpringTestContextManager.prepareTestInstance(SpringTestContextManager.java:50)
	at org.spockframework.spring.SpringInterceptor.interceptSetupMethod(SpringInterceptor.java:42)
	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:28)
	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:87)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:114)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:109)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:147)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:129)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'bean.test.beans.DummyDependency' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.spockframework.runtime.model.FieldMetadata(line=13, name=dummyDependency, ordinal=0)}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:1493)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1104)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:585)
	... 30 more
```

### Environment Information

- **Operating System**: Windows 10
- **Grails Version:** ``3.3.5``
- **JDK Version:** ``javac 1.8.0_102``
- **Container Version (If Applicable):** n/a

### Example Application

- https://github.com/miffels/catfood/tree/master/bean-test

--- a/grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/IntegrationTestGradlePlugin.groovy
+++ b/grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/IntegrationTestGradlePlugin.groovy
@@ -45,15 +45,19 @@ class IntegrationTestGradlePlugin implements Plugin<Project> {
     @CompileStatic
     void apply(Project project) {
         File[] sourceDirs = findIntegrationTestSources(project)
-        if(sourceDirs) {
+        if (sourceDirs) {
             List<File> acceptedSourceDirs = []
             SourceSetContainer sourceSets = SourceSets.findSourceSets(project)
             SourceSet integrationTest = sourceSets.create("integrationTest")
 
-            for(File srcDir in sourceDirs ) {
+            for (File srcDir in sourceDirs) {
                 registerSourceDir(integrationTest, srcDir)
                 acceptedSourceDirs.add srcDir
             }
+
+            File resources = new File(project.projectDir, "grails-app/conf")
+            integrationTest.resources.srcDir(resources)
+
             DependencyHandler dependencies = project.dependencies
             dependencies.add("integrationTestCompile", SourceSets.findMainSourceSet(project).output)
             dependencies.add("integrationTestCompile", SourceSets.findSourceSet(project, SourceSet.TEST_SOURCE_SET_NAME).output)
@@ -82,7 +86,7 @@ class IntegrationTestGradlePlugin implements Plugin<Project> {
 
             integrationTestTask.finalizedBy testReportTask
 
-            if(ideaIntegration) {
+            if (ideaIntegration) {
                 integrateIdea(project, acceptedSourceDirs)
             }
         }

